scripts/sct_deepseg.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+"""
+This command-line tool is the interface for the deepseg API that performs segmentation using deep learning from the
+ivadomed package.
+"""
+
+# TODO: Add link to example image so users can decide wether their images look "close enought" to some of the proposed
+#  models (e.g., mice, etc.).
+# TODO: add test to make sure that all postprocessing flags match the core.DEFAULT dictionary items
+# TODO: Fetch default value (and display) depending on the model that is used.
+
+from __future__ import absolute_import
+
+import sys
+import os
+import argparse
+import colored
+
+import spinalcordtoolbox as sct
+import spinalcordtoolbox.deepseg.core
+import spinalcordtoolbox.deepseg.models
+import spinalcordtoolbox.utils
+
+from sct_utils import init_sct, printv, display_viewer_syntax
+
+
+def get_parser():
+
+    parser = argparse.ArgumentParser(
+        description="Segmentation using deep learning.",
+        add_help=None,
+        formatter_class=sct.utils.SmartFormatter,
+        prog=os.path.basename(__file__).strip(".py"))
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        "-i",
+        help="Image to segment.",
+        metavar=sct.utils.Metavar.file)
+
+    seg = parser.add_argument_group('\nMODELS')
+    seg.add_argument(
+        "-m",
+        help="Model to use. For a description of each available model, please write: 'sct_deepseg -list-models'.",
+        choices=list(sct.deepseg.models.MODELS.keys()))
+    seg.add_argument(
+        "-list-models",
+        action='store_true',
+        help="Display a list of available models.")
```

Comment:
This takes 2.5s to run on my machine. 2.5s to do `ls` of maximum 3 folders seems crazy expensive.

I am going to look deeper into this. I bet there's a bunch of expensive imports that don't really need to happen in here:

```
(venv_sct) $ python -m cProfile -s cumulative scripts/sct_deepseg.py -list-models

--
Spinal Cord Toolbox (git-jca/2626-deepseg-0f537cb59edb3c780f718d28304be45da11d6fe0)

MODEL                    DESCRIPTION
--------------------------------------------------------------------------------
t2star_sc[*]             Cord segmentation on T2*-weighted contrast.
mice_uqueensland_sc      Cord segmentation on mouse MRI. Data from University of Queensland.
mice_uqueensland_gm      Gray matter segmentation on mouse MRI. Data from University of Queensland.

Legend: installed | not installed | default: [*]

         1488696 function calls (1452030 primitive calls) in 2.225 seconds

   Ordered by: cumulative time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
       16    0.000    0.000    2.462    0.154 utils.py:1(<module>)
   2208/1    0.105    0.000    2.233    2.233 {built-in method builtins.exec}
        1    0.000    0.000    2.233    2.233 sct_deepseg.py:6(<module>)
   2215/6    0.011    0.000    2.230    0.372 <frozen importlib._bootstrap>:966(_find_and_load)
   2211/6    0.009    0.000    2.230    0.372 <frozen importlib._bootstrap>:936(_find_and_load_unlocked)
   1778/7    0.008    0.000    2.229    0.318 <frozen importlib._bootstrap>:651(_load_unlocked)
   1555/7    0.004    0.000    2.229    0.318 <frozen importlib._bootstrap_external>:672(exec_module)
  2382/10    0.001    0.000    2.228    0.223 <frozen importlib._bootstrap>:211(_call_with_frames_removed)
        1    0.000    0.000    2.170    2.170 core.py:4(<module>)
   690/85    0.002    0.000    1.654    0.019 {built-in method builtins.__import__}
8092/1585    0.006    0.000    1.581    0.001 <frozen importlib._bootstrap>:997(_handle_fromlist)
       98    0.002    0.000    1.532    0.016 __init__.py:1(<module>)
       24    0.000    0.000    0.706    0.029 __init__.py:3(<module>)
        2    0.001    0.000    0.397    0.199 __init__.py:11(<module>)
```

Reply:
For example, why are there at least 7 calls into Pandas?

```
        7    0.000    0.000    0.000    0.000 {pandas._libs.tslibs.offsets.apply_index_wraps}
```

Reply:
yes, i noticed this as well. Related issue: https://github.com/neuropoly/ivado-medical-imaging/issues/194